### ISSUE TITLE
[feature] Add Alt-Text Tags to Resolve Critical WCAG Accessibility Issues in Web3Modal

### ISSUE THREAD

**Issue by yungzyad** [WalletConnect/web3modal#2698]
### What problem does this new feature solve?

Users that depend on assistive technologies like screen readers are unable to use Web3Modal since icon buttons and other UI elements lack alt-text tags. This minor fix will reduce barriers to entry for millions of web3 users! 

[See here for an accessibility report for the Web3Modal demo](https://axe.deque.com/axe-devtools/user-flows/33e5cc7f-7271-4146-99ae-fa4617cb4753/issues?configNeedsReview=disabled&configBestPractices=disabled&configAccessibilityStandard=wcag21aa&impact=critical&issueIndex=1&ruleId=button-name) on lab.web3modal.com/library/wagmi-all/

### Describe the solution you'd like

Add descriptive `aria-label` tags to icon buttons and other UI elements in Web3Modal, or [another WCAG approved solution](https://dequeuniversity.com/rules/axe/4.10/button-name?application=AxeChrome). 

References to some of the offending UI elements:
**Help Icon Button**
https://github.com/WalletConnect/web3modal/blob/main/packages/scaffold-ui/src/partials/w3m-header/index.ts#L208
**Close Icon Button**
https://github.com/WalletConnect/web3modal/blob/main/packages/scaffold-ui/src/partials/w3m-header/index.ts#L165
**Header Text**
https://github.com/WalletConnect/web3modal/blob/main/packages/scaffold-ui/src/partials/w3m-header/index.ts#L184

**Comment by rtomas** [comment]
Thank you for the valuable feedback!

**Comment by rtomas** [comment]
https://linear.app/walletconnect/issue/APKT-876/[feature]-add-alt-text-tags-to-resolve-critical-wcag-accessibility

**Comment by sleepyqadir** [comment]
Hey @rtomas , I can work on this can you assign this to me?

**Comment by p1kalys** [comment]
Hey @rtomas, I would like to work on the issue, can you please assign it to me if not resolved?

**Comment by VamshikrishDanam** [comment]
Could you please fix These as well, thank you! If we run axe test we see them as critical. 

1. ARIA dialog and alertdialog nodes should have an accessible name
    Description: Ensures every ARIA dialog and alertdialog node has an accessible name
    Help: ARIA dialog and alertdialog nodes should have an accessible name
    Help URL: https://dequeuniversity.com/rules/axe/4.5/aria-dialog-name?application=axeAPI
    Impact: serious
    Tags: cat.aria, best-practice
    Element(s):
    Html:
    <div id=“w3m-modal” role=“alertdialog” aria-modal=“true” class=” w3m-overlay “><div class=“w3m-container” 
     tabindex=“0"></div></div>
    Selector(s):
    w3m-modal, #w3m-modal
    
    Solution -  Add aria-label
    
2. ARIA attributes must conform to valid values
    Description: Ensures all ARIA attributes have valid values
    Help: ARIA attributes must conform to valid values
    Help URL: https://dequeuniversity.com/rules/axe/4.5/aria-valid-attr-value?application=axeAPI
    Impact: critical
    Tags: cat.aria, wcag2a, wcag412
    Element(s):
    Html:
    <div id="w3m-modal" role="alertdialog" aria-modal="true" class="w3m-overlay w3m-active" aria-labelledby="web3- 
    modal" style="opacity: 1;">
    Selector(s):
    w3m-modal, #w3m-modal 
    
    Solution - add aria-hidden value true/false when modal is open and closed respectively.
